src/dialog/dialog.service.ts

```diff
@@ -140,10 +143,10 @@ export class DialogService {
 
 		if (this.dialogRef) {
 			let elementToFocus = this.dialogRef.instance.dialogConfig["previouslyFocusedElement"];
-			if (this.dialogRef.instance.dialogConfig.appendToBody) {
-				this.placeholderService.destroyComponent(this.dialogRef);
-			} else {
+			if (this.dialogRef.instance.dialogConfig.appendInline) {
 				viewContainer.remove(viewContainer.indexOf(this.dialogRef.hostView));
+			} else if (this.placeholderService.hasPlaceholderRef()) {
+				this.placeholderService.destroyComponent(this.dialogRef);
 			}
```

Comment:
Do we need an `else` for the ones that are actually in the `body`?

Reply:
it didn't have one before, so I guess not?